pkg/broker/instance_operations.go

```diff
@@ -462,6 +590,20 @@ func (b Broker) LastOperation(ctx context.Context, instanceID string, details do
 			return
 		}
 
+		retry := false
+		logger.Debugw("Create resources", "plan", p)
+		retry, err = b.postCreateResources(ctx, client, p)
+		if err != nil {
+			break
+		}
+
+		if retry {
```

Comment:
[q] am I right that the `retry` variable affects results in some retry of the logic in the code? I don't see where this variable is used though - could you point please?

Reply:
> // TODO: this retry logic is clunky, come up with something better?
The way I see it `postCreateResources` method returns `retry` as sort of a flag. Which indicates if the same request should be repeated. 

```go
if retry {
	resp.State = domain.InProgress
	resp.Description = "resources are being created"

	break
}
```
Here we break if this flag is `true`.

This solution seems temporary but I haven't yet started thinking of a better alternative.